logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/ProtoContentMapper.kt

```diff
@@ -99,9 +99,8 @@ class ProtoContentMapperImpl : ProtoContentMapper {
         val content = when (val protoContent = genericMessage.content) {
             is GenericMessage.Content.Text -> MessageContent.Text(protoContent.value.content)
             is GenericMessage.Content.Asset -> {
-                if (protoContent.value.uploaded != null) MessageContent.Asset(
-                    MapperProvider.assetMapper().fromProtoAssetMessageToAssetContent(protoContent.value)
-                ) else MessageContent.Unknown // Sometimes backend sends some preview asset messages just with img metadata and no keys or asset id, so we discard it
+                // Backend sends some preview asset messages just with img metadata and no keys or asset id, so we need to overwrite one with the other one
```

Comment:
Maybe it would be better to distinguish them by creating a sealed class instead of passing `AssetContent.RemoteData` with empty data? 🤔

Reply:
I think that's a bit overkill, as this will probably (hopefully) be improved by the backend to accommodate to the way we do it in Android now, i.e, with just one asset message instead of 2